src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/x64/windows/CallArranger.java

```diff
@@ -183,7 +183,10 @@ private static TypeClass classifyValueType(ValueLayout type) {
         } else if(clazz == ArgumentClassImpl.POINTER) {
             return TypeClass.POINTER;
         } else if (clazz == ArgumentClassImpl.SSE) {
-            if (type.attribute(VARARGS_ATTRIBUTE_NAME, String.class).map(Boolean::parseBoolean).orElse(false)) {
+            if (type.attribute(VARARGS_ATTRIBUTE_NAME)
+                    .filter(String.class::isInstance)
```

Comment:
Not convinced by the `filter` here? You get an Optional<Constable>, you do a cast to String and then parse as boolean - why the filtering? If there's an annotation with that name, but different type, I'd prefer to crash than to end up doing something wrong. Same applies elsewhere.